shell/platform/darwin/macos/framework/Source/FlutterViewController_Internal.h

```diff
@@ -2,20 +2,63 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+#include <map>
+#include <unordered_set>
 
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformViews.h"
 #import "flutter/shell/platform/darwin/macos/framework/Source/FlutterView.h"
 
 @interface FlutterViewController ()
 
 // The FlutterView for this view controller.
 @property(nonatomic, readonly, nullable) FlutterView* flutterView;
 
+// NSDictionary maps strings to FlutterPlatformViewFactorys.
+@property(nonnull, nonatomic)
+    NSMutableDictionary<NSString*, NSObject<FlutterPlatformViewFactory>*>* factories;
+
+// A map of platform view ids to views.
+@property(nonatomic) std::map<int, NSView*> platformViews;
+
+// View ids that are going to be disposed on the next present call.
+@property(nonatomic) std::unordered_set<int64_t> platformViewsToDispose;
+
 /**
  * This just returns the NSPasteboard so that it can be mocked in the tests.
  */
 @property(nonatomic, readonly, nonnull) NSPasteboard* pasteboard;
 
+/**
```

Comment:
Create a new class `FlutterPlatformViewController` and move these methods there.